addons/stock_account/models/account_invoice.py

```diff
@@ -111,7 +112,9 @@ def _get_price(self, company_currency, price_unit):
         return self.invoice_id.currency_id.round(price)
 
     def get_invoice_line_account(self, type, product, fpos, company):
-        if company.anglo_saxon_accounting and type in ('in_invoice', 'in_refund') and product and (product.type == 'product' or product.type == 'consu' and product._is_phantom_bom()):
+        purchase_line = self._context.get('purchase_line_id')
```

Comment:
I think you could do an override in `purchase_stock` for this. Follow the same idea than the method `_is_phantom_bom`, which is also a hack.